### PR TITLE
Scrape Artists from Musicbrainz

### PR DESCRIPTION
**Pull request by metavaults**

### CODE REVIEW

**Comment by sacdallago** [comment]
Put someone as reviewer(s) (not us mentors! We are just reviewing these first PRs thoroughly to teach you _how it's done_, next times we will limit ourselves to close PRs if it's not alright <sorry in advance, happens.>)

**Review by sacdallago — DISMISSED** [review]
No scraped data on the repo online! `.gitignore` it and remove it from the PR

File: scrapeArtists/server.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            }
+
+            console.log(url);
+
+            console.log('request started!');
+            
+            request({
+                url: url,
+                headers: {
+                'User-Agent': 'ClassicalApp/1.4.0 ( tumiss@gmail.com )'
```
Comment (kordianbruck) [inline scrapeArtists/server.js]: You should be careful to commit an email address. Github is crawled for such sensitive information

**Comment by sacdallago** [comment]
Updates on this?

**Comment by TimHenkelmann** [comment]
Applied tutor suggestions for improvement:
- Removed package.json and outputfile artistsFromRecordings.json
- added .gitignore and artistsFromRecordings_example.json
- applied changes to server.js:
    - removed server
    - removed email

Ready for review!

**Comment by TimHenkelmann** [comment]
Hi @sacdallago. I have a question for this one. I removed the scraped output (artistsFromRecordings.js) from the PR. However, the server.js script needs to read data from "IDArtistsRecordingsFinal2.json" (an input file which contains scraped data from another script).
How should we proceed with that:
- Remove the input file as well
- Rename it and only show a few entries (then one would also need to change the server.js file)
- Keep the name of the input file and only have it contain a few entries
-  Something completely different

Thanks in advance

File: scrapeArtists/artistsFromRecordings_example.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"table":[{"artistId":"e8c91353-e251-456b-80ad-d10c2cd00be3","aboutartist":{"gender":null,"isnis":[],"id":"e8c91353-e251-456b-80ad-d10c2cd00be3","life-span":{"ended":false,"end":null,"begin":null},"name":"University of Arkansas Chorus","begin_area":null,"type":"Choir","sort-name":"University of Arkansas Chorus","country":null,"area":{"id":"8788d6c2-c779-4be5-ad47-cf0a95e0f2a0","disambiguation":"","iso-3166-2-codes":["US-AR"],"name":"Arkansas","sort-name":"Arkansas"},"end_area":null,"gender-id":null,"disambiguation":"","type-id":"6124967d-7e3a-3eba-b642-c9a2ffb44d94","ipis":[]}},{"artistId":"69efc63a-6bf3-4763-8a30-cf7fd66995f6","aboutartist":{"type-id":"b6e035f4-3ce9-331c-97df-83397230b0df","ipis":[],"area":{"id":"489ce91b-6658-3307-9877-795b68554c98","iso-3166-1-codes":["US"],"disambiguation":"","name":"United States","sort-name":"United States"},"end_area":null,"gender-id":"36d3d30a-839d-3eda-8cb3-29be4384e4a9","disambiguation":"","life-span":{"begin":null,"end":null,"ended":false},"name":"Jack Groh","begin_area":null,"sort-name":"Groh, Jack","type":"Person","country":"US","gender":"Male","isnis":[],"id":"69efc63a-6bf3-4763-8a30-cf7fd66995f6"}}]}
```
Comment (kordianbruck) [inline scrapeArtists/artistsFromRecordings_example.json]: Please don't commit these json files. It only blows up your Git History and makes it slow.

**Review by kordianbruck — CHANGES_REQUESTED** [review]

File: scrapeArtists/artistsFromRecordings_example.json (hunk 2 above)
Comment (kordianbruck) [reply scrapeArtists/artistsFromRecordings_example.json]: Or is it intended?

**Comment by kordianbruck** [comment]
@TimHenkelmann - you should only write files to disk as a cache. Ideally we only start one server / script that will handle all scraping and loading while keeping everything in ram. (We have kinda unlimited Ram)

**Comment by sacdallago** [comment]
Mh, if I understood correctly: this script has a dependency on another script to get data, right?

Natural thing to do would chain these two scripts together, something like:

```
IF NOT dataFromScriptOne exists:
   require scriptOne
   run scriptOne.getData
   run script
ELSE
   run script
```

**Review by metavaults — COMMENTED** [review]
- did all changes according to reviewers

File: scrapeArtists/artistsFromRecordings_example.json (hunk 2 above)
Comment (TimHenkelmann) [reply scrapeArtists/artistsFromRecordings_example.json]: @kordianbruck Yes, this is only an extract of the complete json output file and displays how the data is going to look like

**Comment by kordianbruck** [comment]
@MusicConnectionMachine/group-1 please also review this! Its been waiting for **7 days** - ahhhh :angry:

**Comment by TimHenkelmann** [comment]
Hi @kordianbruck , there are still "requested changes" from your side (probably from the .json file that we had in there at first but now deleted). Could you accept them or comment on that?
Thx!

File: scrapeArtists/server.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    var elemNum = 0;
+    var newObj = {
+      table: []
+    };
+
+    var finalArray = arrayofObj.map(function (artistai) {
+      return artistai.artistId;
+    });
+
+    setInterval(function(){
```
Comment (kordianbruck) [inline scrapeArtists/server.js:20]: Why are you using a setInterval here?

**Review by kordianbruck — CHANGES_REQUESTED** [review]
Last thing.

File: scrapeArtists/server.js (hunk 3 above)
Comment (metavaults) [reply scrapeArtists/server.js:20]: In order to get 100% of results, we cannot query MusicBrainz faster than every second. 
Reference: https://musicbrainz.org/doc/XML_Web_Service/Rate_Limiting

Comment (kordianbruck) [reply scrapeArtists/server.js:20]: Ugly solution. Might want to reconsider this in the future.

**Review by kordianbruck — APPROVED** [review]

File: scrapeArtists/server.js (hunk 3 above)
Comment (TimHenkelmann) [reply scrapeArtists/server.js:20]: Alright, we will try to find other approaches for this in the future!

Comment (kordianbruck) [reply scrapeArtists/server.js:20]: Yea, don't worry about it now.

**Pull request merged**